dask/tests/test_distributed.py

```diff
@@ -606,6 +606,7 @@ def f(d, a):
     await c.compute(df.map_partitions(f, arr, meta=df._meta))
 
 
+@pytest.mark.network
```

Comment:
@jrbourbeau 

This one is a stretch. With all network connections killed, the error message I'm getting without the decorator is in the details below. It's not actually an error -- just a failure due to a raised warning.

<details>
</pre>
>       with distributed.LocalCluster(
            scheduler_port=0,
            dashboard_address=":0",
            asynchronous=False,
            n_workers=1,
            nthreads=1,
            processes=False,
        ) as cluster:

dask/tests/test_distributed.py:634:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/deploy/local.py:236: in __init__
    super().__init__(
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/deploy/spec.py:260: in __init__
    self.sync(self._start)
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/utils.py:309: in sync
    return sync(
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/utils.py:376: in sync
    raise exc.with_traceback(tb)
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/utils.py:349: in f
    result = yield future
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/tornado/gen.py:762: in run
    value = future.result()
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/deploy/spec.py:293: in _start
    self.scheduler = await self.scheduler
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/core.py:299: in _
    await self.start()
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/scheduler.py:4180: in start
    await self.listen(
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/core.py:437: in listen
    listener = await listen(
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/comm/core.py:366: in listen
    return backend.get_listener(loc, handle_comm, deserialize, **kwargs)
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/comm/inproc.py:343: in get_listener
    return InProcListener(loc, handle_comm, deserialize)
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/comm/inproc.py:250: in __init__
    self.address = address or self.manager.new_address()
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/comm/inproc.py:63: in new_address
    return "%s/%d/%s" % (self.ip, os.getpid(), next(self.addr_suffixes))
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/comm/inproc.py:39: in ip
    self._ip = get_ip()
../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/utils.py:155: in get_ip
    return _get_ip(host, port, family=socket.AF_INET)
cytoolz/functoolz.pyx:476: in cytoolz.functoolz._memoize.__call__
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

host = '8.8.8.8', port = 80, family = <AddressFamily.AF_INET: 2>

    @toolz.memoize
    def _get_ip(host, port, family):
        # By using a UDP socket, we don't actually try to connect but
        # simply select the local address through which *host* is reachable.
        sock = socket.socket(family, socket.SOCK_DGRAM)
        try:
            sock.connect((host, port))
            ip = sock.getsockname()[0]
            return ip
        except OSError as e:
>           warnings.warn(
                "Couldn't detect a suitable IP address for "
                "reaching %r, defaulting to hostname: %s" % (host, e),
                RuntimeWarning,
            )
E           RuntimeWarning: Couldn't detect a suitable IP address for reaching '8.8.8.8', defaulting to hostname: [Errno 51] Network is unreachable

../../mambaforge/envs/dask-mypy/lib/python3.9/site-packages/distributed/utils.py:135: RuntimeWarning
</pre>
</details>

Reply:
Ah, yeah, that is what's happening. I'd just silence that warning in `setup.cfg` by adding it to `filterwarnings` rather than doing anything specific to this test. That warning is innocuous in our tests.

Reply:
+1 for _not_ elevating this specific warning to an error. @phobson, as @jcrist mentioned, adding a section like

```
# With no network access, distributed raises a warning when detecting local IP address
ignore:Couldn't detect a suitable IP address:RuntimeWarning:distributed
```

here

https://github.com/dask/dask/blob/0b36d7fcaf54ee9a78fff4b07f124cb0c8741cdf/setup.cfg#L46-L54

will allow us to not raise an unnecessary error for these cases

Reply:
thanks @jcrist & @jrbourbeau. I modified the setup.cfg per the suggestion and tests passed locally w/o a network. Just pushed a commit with that change and reverting the change to test_distributed.py